### DISCUSSION TITLE
Managed to failsafe really close. Any ideas?

### DISCUSSION THREAD

**Discussion by Strat00s** [ExpressLRS/ExpressLRS#1794] [General]
First time flying with ELRS and I just wonder if I am cursed.

I have Tinyhawk Freestyle and I love to just cruise around with it. I can get up to 10min flight time with it, so I always try to fly as far as I can. But I always failsafe about 300 meters out, always before my video cuts out. So I wanted to try ELRS as I always read how good the RC link is. I bought JHEMCU SP24S to put on my Tinyhawk and Happymodel ES24TX Pro (set to 50mw) to put on my QX7 radio. Flashed both with the latest firmware, tested everything on the bench, and went flying. Everything seemed fine, tried some very close flying and everything looked good. Then I tried my usual route around a field at around 7 meters high. After getting roughly 240m away from me (checked on a map), it just failsafed (as always, before the video went out). Just as I found the drone it started to rain, so I couldn't test it further.

Guys, what the hell? Am I seriously so cursed, that even with ELRS I just can't get the range? And yes, I know. There isn't much information (as I flew once so far with it). But still, does anyone have any idea what can I do so next time doesn't end like this? And no, I didn't have dynamic power and I probably will use that from now on.

**Comment by SunjunKim** [comment]
Ouch... JHEMCU is a known cursed RX manufacturer with severe frequency offset. Stay away from it. 
https://github.com/ExpressLRS/ExpressLRS/wiki/Crystal-Oscillator-(XO)-Frequency-Error

You can test the matching between your TX/RX pair with the instruction introduced in the "Measuring relative XO error between a TX-RX pair" section.

**Reply by Strat00s** [reply]
Should I stay away from their 868MHz receivers too?

**Reply by Strat00s** [reply]
Ok. After flashing both, the RSNR is jumping between -70 and -72. So yea, it is really bad. Is there anything that can be done?

**Reply by SunjunKim** [reply]
It's actually better than I expected. It's not ideal but not bad enough to knock your quad out of the sky. However, it might still be vulnerable on the temperature stability issue.

Could you take a detailed picture of your receiver setup (the placement of RX-- should not be on any heat source nearby (like VTX, ESC, FC, etc.)? Also, you may heat the receiver with a hairdryer and check the RSNR field for temperature-related deviation.

You may also try "capacitor removal" surgery if it has some extra caps nearby the 52M crystal units. If you could post the detailed picture of your receiver (both sides) then I can help with which parts should be removed.

**Reply by SunjunKim** [reply]
https://media.discordapp.net/attachments/798006228450017290/1008232318135193611/SimoneFrascarelli.jpg?width=604&height=676

FYI they're the capacitors you'll have to remove, if the board layout is same to this picture.

**Reply by Strat00s** [reply]
Looks to be the same receiver, but both caps are already missing from mine. I didn't remove them, so maybe JHEMCU is somewhat trying to fix their design. They don't seem to be torn off either.
The receiver was placed between the AIO (bellow one of the ESCs) and the drone frame (third image). It was also in a heat-shrink tube.
<details>
  <summary>Spoiler warning</summary>

  ![2014_0101_000539_003](https://user-images.githubusercontent.com/24775036/186848690-38641c6b-cbdd-4bbd-b8d9-ad0fa05b0bb0.JPG)
  ![2014_0101_000639_005](https://user-images.githubusercontent.com/24775036/186848694-4c7773c1-2d89-4581-a6f0-2af08823c7ca.JPG)
![20220826_094422](https://user-images.githubusercontent.com/24775036/186850522-74b27015-cb7a-46c7-852d-548f9ce49318.jpg)

</details>

**Comment by Strat00s** [comment]
Tried it again today and it is extremely unreliable. Flew about 400ish meters around me and no problem. Got a low RSSI warning a few times, but RQly was pretty solid (although it dipped once to around 76). After I changed the batteries, I wanted to continue flying in about the same range, but it once again failsafed at around 250m away. The most interesting thing is that RQly was 96 at the time of failsafe (TQly on the other hand was getting worse and worse).
After the failsafe, it wouldn't even connect until I was like 5 meters away from it (it did land in a field of some greenery tho).
So yea. No idea what is going on...

**Reply by commanderguy3001** [reply]
did you do the [bench test](https://expresslrs.org/2.0/quick-start/pre-1stflight/#bench-test)? If so, what were your results?

**Reply by Strat00s** [reply]
I am getting around -35db. So pretty much exactly what is written to be good.

**Reply by SunjunKim** [reply]
could you post some dvr or log if you have any?

**Reply by Strat00s** [reply]
Here is a log from the last failsafe.
[Freestyle-2022-08-27-183848.csv](https://github.com/ExpressLRS/ExpressLRS/files/9439284/Freestyle-2022-08-27-183848.csv)

**Reply by SunjunKim** [reply]
> Here is a log from the last failsafe. [Freestyle-2022-08-27-183848.csv](https://github.com/ExpressLRS/ExpressLRS/files/9439284/Freestyle-2022-08-27-183848.csv)

By looking at the log, all the values insist that the RF strength itself was dropping gradually from 18:40:50, all the way to the complete telemetry lost. The TPWR is stepping up to the 1 W, and because TX/RX has asymmetric rf power, the RX probably got some more packets after the point of the log was stopped. You probably experienced a "telemetry lost" alarm a few seconds ahead of the actual failsafe.

So it's all about RF signal quality. Did you have a clear line of sight (LOS) when it failsafed? Were there any obstacles? or, were you flying with some other drones?

**Reply by Strat00s** [reply]
> > Here is a log from the last failsafe. [Freestyle-2022-08-27-183848.csv](https://github.com/ExpressLRS/ExpressLRS/files/9439284/Freestyle-2022-08-27-183848.csv)
> 
> By looking at the log, all the values insist that the RF strength itself was dropping gradually from 18:40:50, all the way to the complete telemetry lost. The TPWR is stepping up to the 1 W, and because TX/RX has asymmetric rf power, the RX probably got some more packets after the point of the log was stopped. You probably experienced a "telemetry lost" alarm a few seconds ahead of the actual failsafe.
> 
> So it's all about RF signal quality. Did you have a clear line of sight (LOS) when it failsafed? Were there any obstacles? or, were you flying with some other drones?

No drones, clear LOS. And yes, I can see that I lost telemetry (it even said it), but why did I also lose control?

**Reply by SunjunKim** [reply]
The primary reason is your RF signal got weaker enough until the failsafe obviously, but the question is what's the cause of that.
It might be the poor quality RX XO offset gone too far, or some unknown RF source was nuking around your drone at that time, etc. etc.

What I can suggest is, to try replace your RX with something more quality one, like HappyModel EP2, RadioMaster RP2.

**Reply by Strat00s** [reply]
> The primary reason is your RF signal got weaker enough until the failsafe obviously, but the question is what's the cause of that. It might be the poor quality RX XO offset gone too far, or some unknown RF source was nuking around your drone at that time, etc. etc.
> 
> What I can suggest is, to try replace your RX with something more quality one, like HappyModel EP2, RadioMaster RP2.

Hold on. Isn't TQly the quality of transmitted packets **from** the receiver and RQly the quality of received packets **to** the receiver?

**Reply by SunjunKim** [reply]
Yes it is. TQly: RX -> TX, RQly: TX -> RX.

Why you get high RQly in the last part of your telemetry logging is just because your telemetry is dropped at that time.
In reality, your RQly was probably also dropped until the point of failsafe after that, but it couldn’t be logged because of you lost the telemetry way before that event.

**Reply by SunjunKim** [reply]
If you also have the DVR (with OSD of RSSI dbm and LQ) then we can diagnose the incident in more detail.

**Reply by Strat00s** [reply]
> Yes it is. TQly: RX -> TX, RQly: TX -> RX.
> 
> Why you get high RQly in the last part of your telemetry logging is just because your telemetry is dropped at that time. In reality, your RQly was probably also dropped until the point of failsafe after that, but it couldn’t be logged because of you lost the telemetry way before that event.

Right, that makes sense. Didn't think of that.



> If you also have the DVR (with OSD of RSSI dbm and LQ) then we can diagnose the incident in more detail.

Sadly I don't have a DVR.

Also, I already ordered the EP2 receiver. So I am gonna try that tomorrow and report back. But I guess that that one will work just fine.

**Comment by Strat00s** [comment]
Tested the EP2 today. With the debug firmware RSNR was -2, so it's a perfect receiver-transmitter pair and it shows. The line quality never dropped below 96%. I am never touching the JHEMCU again...
